format.go

```diff
@@ -143,6 +144,17 @@ func NewHeader(buf []byte) (*Header, error) {
 	return h, nil
 }
 
+func (h *Header) CalculateCheckSum(key, value string) uint32 {
+	// encode header
+	headerBuf := new(bytes.Buffer)
+	h.EncodeHeader(headerBuf)
+	// encode kv
+	kvBuf := append([]byte(key), []byte(value)...)
+
+	buf := append(headerBuf.Bytes()[4:], kvBuf...)
+	return crc32.ChecksumIEEE(buf)
```

Comment:
@avinassh , I have considered all fields only, except the crc field itself while calculating the checksum.

Reply:
ahh, I saw only `CalculateCheckSum(key, value string)`, assumed something else. Right, you do use other fields. 

but this looks buggy? Because you are calling EncodeHeader, which includes checksum (which could be empty bytes for now)

> err := binary.Write(buf, binary.LittleEndian, &h.CheckSum)

I would suggest moving `CalculateCheckSum` to be part of Record struct and make it idempotent, it should return same value whether header already contains checksum or not

Reply:
Yeah, I just thought of reusing the `EncodeHeader` and picking up bytes from 4th index(effectively excluding CheckSum field). But I get your point, have made the change!